client.go

```diff
@@ -1403,12 +1403,12 @@ func (cli *Client) GetDownloadURL(mxcURL id.ContentURI) string {
 	return cli.BuildURLWithQuery(MediaURLPath{"v3", "download", mxcURL.Homeserver, mxcURL.FileID}, map[string]string{"allow_redirect": "true"})
 }
 
-func (cli *Client) Download(ctx context.Context, mxcURL id.ContentURI) (io.ReadCloser, error) {
+func (cli *Client) Download(ctx context.Context, mxcURL id.ContentURI) (*http.Response, error) {
 	resp, err := cli.download(ctx, mxcURL)
 	if err != nil {
 		return nil, err
 	}
-	return resp.Body, nil
+	return resp, nil
```

Comment:
No point in having this proxy function anymore, can just rename `download` to `Download`